src/GafferSceneUI/TranslateTool.cpp

```diff
@@ -174,12 +231,60 @@ bool TranslateTool::dragMove( const GafferUI::Gadget *gadget, const GafferUI::Dr
 	return true;
 }
 
-bool TranslateTool::dragEnd()
+bool TranslateTool::handleDragEnd()
 {
 	TransformTool::dragEnd();
 	return false;
 }
 
+bool TranslateTool::buttonPress( const GafferUI::ButtonEvent &event )
+{
+	if( event.buttons != ButtonEvent::Left
+		|| !activePlug()->getValue()
+		|| !getTargetedMode()
+	) {
+		return false;
+	}
+
+	// In targeted mode, we teleport the selection to the clicked point.
+	//
+	// Our prescribed behaviour is to move the bbox center of the selection
+	// to the clicked point. If multiple locations are selected, their combined
+	// bounds should be used, so they retain their existing relative spacing.
+	//
+	// We always return true to prevent the SelectTool defaults.
+
+	GafferScene::ScenePlug::ScenePath _;
+	Imath::V3f targetPos;
+
+	const SceneView *sv = static_cast<const SceneView *>( view() );
+	const Gadget *g = sv->viewportGadget()->getPrimaryChild();
+	const SceneGadget* sg = static_cast<const SceneGadget *>( g );
+	if( !sg->objectAt( event.line, _, targetPos ) )
+	{
+		return true;
+	}
+
+	Box3f selectionCentroids;
+	for( const auto &s : selection() )
+	{
+		Context::Scope scopedContext( s.context.get() );
+
+		const M44f worldTransform = s.upstreamScene->fullTransform( s.upstreamPath );
```

Comment:
This is in the world space of `upstreamScene`, which is _not_ the same as the world space of the Viewer. I think here you want to use `s.scene` and `s.context`. Try this scene to see what I mean - the Group adds a rotation between the viewer and the upstream scene, and breaks the target mode :

```
import Gaffer
import GafferArnold
import GafferScene
import IECore
import imath

Gaffer.Metadata.registerValue( parent, "serialiser:milestoneVersion", 0, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:majorVersion", 55, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:minorVersion", 0, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:patchVersion", 0, persistent=False )

__children = {}

__children["Sphere"] = GafferScene.Sphere( "Sphere" )
parent.addChild( __children["Sphere"] )
__children["Sphere"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Plane"] = GafferScene.Plane( "Plane" )
parent.addChild( __children["Plane"] )
__children["Plane"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["SpotLight"] = GafferArnold.ArnoldLight( "SpotLight" )
parent.addChild( __children["SpotLight"] )
__children["SpotLight"].loadShader( "spot_light" )
__children["SpotLight"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"] = GafferScene.Group( "Group" )
parent.addChild( __children["Group"] )
__children["Group"]["in"].addChild( GafferScene.ScenePlug( "in1", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"]["in"].addChild( GafferScene.ScenePlug( "in2", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"]["in"].addChild( GafferScene.ScenePlug( "in3", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Sphere"]["transform"]["translate"].setValue( imath.V3f( -5.7313008e-07, -1.14626016e-06, -18.0306053 ) )
__children["Sphere"]["__uiPosition"].setValue( imath.V2f( -14.3500004, -0.650000036 ) )
__children["Plane"]["transform"]["translate"].setValue( imath.V3f( 1.70846891, 0.18131198, -1.82803655 ) )
__children["Plane"]["transform"]["scale"].setValue( imath.V3f( 34.1658363, 34.1658363, 34.1658363 ) )
__children["Plane"]["__uiPosition"].setValue( imath.V2f( -1.35000038, -0.650000036 ) )
__children["SpotLight"]["name"].setValue( 'spotLight' )
__children["SpotLight"]["transform"]["translate"].setValue( imath.V3f( 4.61245719e-07, 9.22491438e-07, 14.5107365 ) )
__children["SpotLight"]["__uiPosition"].setValue( imath.V2f( 11.6499996, -0.74999994 ) )
__children["Group"]["in"]["in0"].setInput( __children["Sphere"]["out"] )
__children["Group"]["in"]["in1"].setInput( __children["Plane"]["out"] )
__children["Group"]["in"]["in2"].setInput( __children["SpotLight"]["out"] )
__children["Group"]["transform"]["rotate"].setValue( imath.V3f( 0, 90, 0 ) )
__children["Group"]["__uiPosition"].setValue( imath.V2f( 0.150223017, -8.9140625 ) )


del __children
```

Note also that the Transform node has a `space` plug that we need to test with. In theory it should all be taken care of automatically by `Translation` via `Selection::sceneToTransformSpace()`, but we should test once the wrong-world-space issue is fixed.

Reply:
The test case seems to work in https://github.com/GafferHQ/gaffer/commit/e70249fba5e6137ea8b40d8038282b554ad96d34, need to fully test different spaces still.

Reply:
Tested different spaces in the Transform node interactively here : all good.